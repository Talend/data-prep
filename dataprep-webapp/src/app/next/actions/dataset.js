import { REDIRECT_WINDOW } from '../constants/actions';

function open(event, { id }) {
	return {
		type: REDIRECT_WINDOW,
		payload: {
			url: `${window.location.origin}/#/playground/dataset?datasetid=${id}`,
		},
	};
}

export default {
	open,
};
