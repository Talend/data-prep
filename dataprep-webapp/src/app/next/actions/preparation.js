import {
	CANCEL_RENAME_PREPARATION,
	OPEN_FOLDER,
	PREPARATION_DUPLICATE,
	RENAME_PREPARATION,
	SET_TITLE_EDITION_MODE,
	FETCH_PREPARATIONS,
	OPEN_PREPARATION_CREATOR,
} from '../constants';

function openPreparation(event, { id, type }) {
	switch (type) {
	case 'folder':
		return {
			type: OPEN_FOLDER,
			id,
			cmf: {
				routerPush: `/preparations/${id}`,
			},
		};
	case 'preparation':
		/* TODO
		- get current url
		- pass it in the url as 'return' query param
		- playground must redirect to this return param on close
		*/
		window.location.href = `${window.location.origin}/#/playground/preparation?prepid=${id}`;
		break;
	default:
		break;
	}
}

function fetchAll() {
	// TODO [NC]: folderId
	return { type: FETCH_PREPARATIONS };
}

function duplicate(event, { model }) {
	return {
		type: PREPARATION_DUPLICATE,
		payload: {
			id: model.id,
		},
	};
}

function rename(event, data) {
	return {
		type: RENAME_PREPARATION,
		payload: {
			id: data.model.id,
			name: data.value,
		},
	};
}

function cancelRename(event, { id }) {
	return {
		type: CANCEL_RENAME_PREPARATION,
		payload: id,
	};
}

function setTitleEditionMode(event, { model }) {
	return {
		type: SET_TITLE_EDITION_MODE,
		payload: model.id,
	};
}

function openCreator() {
	return { type: OPEN_PREPARATION_CREATOR };
}

export default {
	openPreparation,
	fetchAll,
	duplicate,
	rename,
	cancelRename,
	setTitleEditionMode,
	openCreator,
};
