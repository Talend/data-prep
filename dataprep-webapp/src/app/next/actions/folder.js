import { OPEN_FOLDER } from '../constants/actions';

function open(event, { id }) {
	return {
		type: OPEN_FOLDER,
		id,
		cmf: {
			routerPush: `/preparations/${id}`,
		},
	};
}

export default {
	open,
};
