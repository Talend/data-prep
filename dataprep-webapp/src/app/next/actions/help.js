import { OPEN_ABOUT } from '../constants/actions';

function openAbout() {
	return { type: OPEN_ABOUT };
}

export default {
	openAbout,
};
