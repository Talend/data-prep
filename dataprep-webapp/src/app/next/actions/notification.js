import { SUCCESS_NOTIFICATION, ERROR_NOTIFICATION, WARNING_NOTIFICATION } from '../constants/actions';

function success(event, { title, message }) {
	return {
		type: SUCCESS_NOTIFICATION,
		payload: {
			title,
			message,
		},
	};
}
function error(event, { title, message }) {
	return {
		type: ERROR_NOTIFICATION,
		payload: {
			title,
			message,
		},
	};
}
function warning(event, { title, message }) {
	return {
		type: WARNING_NOTIFICATION,
		payload: {
			title,
			message,
		},
	};
}

export default {
	success,
	error,
	warning,
};
