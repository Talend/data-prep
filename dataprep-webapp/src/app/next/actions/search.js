import { SEARCH, SEARCH_SELECT, SEARCH_RESET } from '../constants/actions';

function start({ target }) {
	return { type: SEARCH, payload: target.value };
}

function select(event, { itemIndex, sectionIndex }) {
	return { type: SEARCH_SELECT, payload: { sectionIndex, itemIndex } };
}

function reset() {
	return { type: SEARCH_RESET };
}

export default {
	start,
	select,
	reset,
};
