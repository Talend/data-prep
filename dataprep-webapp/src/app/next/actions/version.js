import { FETCH_VERSION } from '../constants/actions';

function fetch() {
	return { type: FETCH_VERSION };
}

export default {
	fetch,
};
