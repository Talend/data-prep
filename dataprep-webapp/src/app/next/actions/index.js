import { default as redirect } from './redirect';
import { default as preparation } from './preparation';
import { default as help } from './help';
import { default as version } from './version';

export default {
	redirect,
	preparation,
	help,
	version,
};
