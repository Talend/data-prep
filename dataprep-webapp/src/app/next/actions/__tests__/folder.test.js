import actions from '../folder';


describe('Folder action', () => {
	it('should create an open action', () => {
		const event = {};
		const payload = {
			id: 42,
		};
		const action = actions.open(event, payload);

		expect(action).toMatchSnapshot();
	});
});
