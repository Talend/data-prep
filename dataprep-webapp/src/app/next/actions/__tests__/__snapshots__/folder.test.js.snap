// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder action should create an open action 1`] = `
Object {
  "cmf": Object {
    "routerPush": "/preparations/42",
  },
  "id": 42,
  "type": "TDP_OPEN_FOLDER",
}
`;
